src/krb5/_creds_mit.pyx

```diff
@@ -17,7 +20,19 @@ from krb5._principal cimport Principal
 cdef extern from "python_krb5.h":
     """
 #if defined(HEIMDAL_XFREE)
-#error "Heimdal implementation of krb5_get_validated_creds() does not work"
+#error "Heimdal implementation does not support MIT-specific calls:"
+#error " krb5_get_validated_creds()"
+#error " krb5_get_etype_info()"
+#error " krb5_marshal_credentials()"
+#error " krb5_unmarshal_credentials()"
+#else
+    void pykrb5_free_data(
```

Comment:
Is there a reason why we need this? Seems like we can just call `krb5_free_data` directly.

Reply:
correct me if I'm wrong but `krb5_data_free()` compiles in only when in Heimdal mode, it will not be available in MIT for which I am addressing the whole change

Reply:
indeed, not needed